src/Domain/CoordinateMaps/Composition.cpp

```diff
@@ -350,7 +358,11 @@ template <typename Frames, size_t Dim, size_t... Is>
 bool Composition<Frames, Dim, std::index_sequence<Is...>>::is_equal_to(
     const CoordinateMapBase<SourceFrame, TargetFrame, Dim>& other) const {
   const auto& cast_of_other = dynamic_cast<const Composition&>(other);
-  bool result = true;
+  bool result =
+      function_of_time_names_ == cast_of_other.function_of_time_names_;
+  if (not result) {
+    return false;
+  }
   EXPAND_PACK_LEFT_TO_RIGHT(
       (result = result and (*get<Is>(maps_) == *get<Is>(cast_of_other.maps_))));
   return result;
```

Comment:
[optional] Not new in this PR, but this is set of statements is `return ((function_of_time_names_ == cast_of_other.function_of_time_names_) and ... and (*get<Is>(maps_) == *get<Is>(cast_of_other.maps_)));`.